Bug: io.ktor.network.selector.InterestSuspensionsMap cannot initialize in some locales.

I have been looking at this error in my application for a few days.
```
java.lang.ExceptionInInitializerError
at io.ktor.network.selector.SelectableBase.<init>(Selectable.kt:42)
at io.ktor.network.sockets.NIOSocketImpl.<init>(NIOSocket.kt:22)
at io.ktor.network.sockets.SocketImpl.<init>(SocketImpl.kt:15)
at io.ktor.network.sockets.TcpSocketBuilder.connect(Builders.kt:106)
at io.ktor.client.engine.cio.ConnectionFactory.connect(ConnectionFactory.kt:2099)
at io.ktor.client.engine.cio.Endpoint$connect$$inlined$repeat$lambda$1.invokeSuspend(Endpoint.kt:143)
at io.ktor.client.engine.cio.Endpoint$connect$$inlined$repeat$lambda$1.invoke(Endpoint.kt)
at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturnIgnoreTimeout(Undispatched.kt:102)
at kotlinx.coroutines.TimeoutKt.setupTimeout(Timeout.kt:78)
at kotlinx.coroutines.TimeoutKt.access$setupTimeout(Timeout.kt:1)
at kotlinx.coroutines.TimeoutKt.withTimeoutOrNull(Timeout.kt:57)
at io.ktor.client.engine.cio.Endpoint.connect(Endpoint.kt:143)
at io.ktor.client.engine.cio.Endpoint$makeDedicatedRequest$1.invokeSuspend(Endpoint.kt:90)
at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
at java.lang.Thread.run(Unknown Source)
Caused by: java.lang.RuntimeException: java.lang.NoSuchFieldException: wrıteHandlerReference
at java.util.concurrent.atomic.AtomicReferenceFieldUpdater$AtomicReferenceFieldUpdaterImpl.<init>(Unknown Source)
at java.util.concurrent.atomic.AtomicReferenceFieldUpdater.newUpdater(Unknown Source)
at io.ktor.network.selector.InterestSuspensionsMap.<clinit>(InterestSuspensionsMap.kt:65)
... 18 more
Caused by: java.lang.NoSuchFieldException: wrıteHandlerReference
at java.lang.Class.getDeclaredField(Unknown Source)
at java.util.concurrent.atomic.AtomicReferenceFieldUpdater$AtomicReferenceFieldUpdaterImpl$1.run(Unknown Source)
at java.util.concurrent.atomic.AtomicReferenceFieldUpdater$AtomicReferenceFieldUpdaterImpl$1.run(Unknown Source)
at java.security.AccessController.doPrivileged(Native Method)
... 21 more
```

The problem is: `java.lang.NoSuchFieldException: wrıteHandlerReference`
This is not the name of the field, the field's name is `writeHandlerReference`. It's a subtle difference that happens in the Turkish locale (no others that I am aware of).

The cause is on line 65:
`AtomicReferenceFieldUpdater.newUpdater(InterestSuspensionsMap::class.java, CancellableContinuation::class.java, "${interest.name.toLowerCase()}HandlerReference") as AtomicReferenceFieldUpdater<InterestSuspensionsMap, CancellableContinuation<Unit>?>`

It uses this expression to determine the field name associated with each `SelectInterest` constant.

--- a/ktor-network/jvm/src/io/ktor/network/selector/InterestSuspensionsMap.kt
+++ b/ktor-network/jvm/src/io/ktor/network/selector/InterestSuspensionsMap.kt
@@ -62,7 +62,13 @@ class InterestSuspensionsMap {
     companion object {
         @Suppress("UNCHECKED_CAST")
         private val updaters = SelectInterest.AllInterests.map { interest ->
-            AtomicReferenceFieldUpdater.newUpdater(InterestSuspensionsMap::class.java, CancellableContinuation::class.java, "${interest.name.toLowerCase()}HandlerReference") as AtomicReferenceFieldUpdater<InterestSuspensionsMap, CancellableContinuation<Unit>?>
+            val property = when (interest) {
+                SelectInterest.READ -> InterestSuspensionsMap::readHandlerReference
+                SelectInterest.WRITE -> InterestSuspensionsMap::writeHandlerReference
+                SelectInterest.ACCEPT -> InterestSuspensionsMap::acceptHandlerReference
+                SelectInterest.CONNECT -> InterestSuspensionsMap::connectHandlerReference
+            }
+            AtomicReferenceFieldUpdater.newUpdater(InterestSuspensionsMap::class.java, CancellableContinuation::class.java, property.name) as AtomicReferenceFieldUpdater<InterestSuspensionsMap, CancellableContinuation<Unit>?>
         }.toTypedArray()
 
         private fun updater(interest: SelectInterest): AtomicReferenceFieldUpdater<InterestSuspensionsMap, CancellableContinuation<Unit>?> = updaters[interest.ordinal]

